### PR TITLE
Fix Graphite Scaler doesn't properly handle null responses #2944

### LINKED ISSUE #2944: Graphite Scaler doesn't properly handle null responses
### Report

In some configurations, it's possible for Graphite to return null for its latest datapoint. This gets erroneously interpeted as a zero.

```
curl 'http://graphite' 2> /dev/null |jq
[
  {
    "target": "sumSeries(metric)",
    "tags": {
      "name": "metric",
      "aggregatedBy": "sum"
    },
    "datapoints": [
      [
        15323331,
        1651092420
      ],
      [
        null,
        1651092480
      ]
    ]
  }
] 


kubectl get --raw '/apis/external.metrics.k8s.io/v1beta1/namespaces/namespace/metric?labelSelector=app=app' | jq
{
  "kind": "ExternalMetricValueList",
  "apiVersion": "external.metrics.k8s.io/v1beta1",
  "metadata": {},
  "items": [
    {
      "metricName": "metric",
      "metricLabels": null,
      "timestamp": "2022-04-27T20:47:57Z",
      "value": "0" # <<< mis-interpreting null as zero
    }
  ]
} 

```

### Expected Behavior

Nulls are tolerated with the most recent non-null datapoint being used instead.
Nulls should not be mis-interpeted as a zero value

### Actual Behavior

Null graphite responses cause apps to erroneously scale down.

### Steps to Reproduce the Problem

1. Create a graphite scaler
2. Allow the query to return a null value
3. Value is considered as a zero

### Logs from KEDA operator

_No response_

### KEDA Version

_No response_

### Kubernetes Version

_No response_

### Platform

_No response_

### Scaler Details

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by bpinske**
Do not mis-interpet null as zero
Use the latest non-null returned value in a query response

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2944

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 
 ## Unreleased
 
+- **Graphite Scaler**  Use the latest non-null datapoint returned by query.
```
Comment (tomkerkhove) [inline CHANGELOG.md:35]: Can you move this to improvements please?

File: pkg/scalers/graphite_scaler.go [hunk 2]
```diff
@@ -201,10 +201,14 @@ func (s *graphiteScaler) executeGrapQuery(ctx context.Context) (float64, error)
 		return 0, nil
 	}
 
-	latestDatapoint := len(result[0].Datapoints) - 1
-	datapoint := result[0].Datapoints[latestDatapoint][0]
+	// Return the most recent non-null datapoint
+	for i := len(result[0].Datapoints) - 1; i >= 0; i-- {
```
Comment (tomkerkhove) [inline pkg/scalers/graphite_scaler.go:205]: Can we add a test for this?

Comment (JorTurFer) [reply pkg/scalers/graphite_scaler.go:205]: +1

Comment (bpinske) [reply pkg/scalers/graphite_scaler.go:205]: Added

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks for this fix ❤️

**Pull request merged**